lib/main.dart

```diff
@@ -62,49 +65,27 @@ Future<void> main() async {
 
       return true;
     };
-
-    await setDefaultMinimumWindowSize();
-
-    await CakeHive.close();
-
-    await initializeAppConfigs();
+    await initializeAppAtRoot();
 
     runApp(App());
-
     isAppRunning = true;
   }, (error, stackTrace) async {
     if (!isAppRunning) {
       runApp(
-        MaterialApp(
-          debugShowCheckedModeBanner: false,
-          scrollBehavior: AppScrollBehavior(),
-          home: Scaffold(
-            body: SingleChildScrollView(
-              child: Container(
-                margin: EdgeInsets.only(top: 50, left: 20, right: 20, bottom: 20),
-                child: Column(
-                  children: [
-                    Text(
-                      'Error:\n${error.toString()}',
-                      style: TextStyle(fontSize: 22),
-                    ),
-                    Text(
-                      'Stack trace:\n${stackTrace.toString()}',
-                      style: TextStyle(fontSize: 16),
-                    ),
-                  ],
-                ),
-              ),
-            ),
-          ),
-        ),
+        TopLevelErrorWidget(error: error, stackTrace: stackTrace),
       );
     }
 
     ExceptionHandler.onError(FlutterErrorDetails(exception: error, stack: stackTrace));
   });
 }
 
+Future<void> initializeAppAtRoot({bool reInitializing = false}) async {
+  if (reInitializing) await setDefaultMinimumWindowSize();
```

Comment:
this was set when app is initialized normally, why change it to be only when restore?

Reply:
Ohh, error from my end.

It's supposed to be inverse. I'll adjust it now.

The check is, if it's not a restore flow, don't call the defaults again.

Reply:
Done